pkg/client/download_sbom.go

```diff
@@ -0,0 +1,57 @@
+package client
+
+import (
+	"context"
+
+	"github.com/buildpacks/lifecycle/layers"
+	"github.com/buildpacks/lifecycle/platform"
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/pack/pkg/dist"
+	"github.com/buildpacks/pack/pkg/image"
+)
+
+type DownloadSBOMOptions struct {
+	Daemon         bool
+	DestinationDir string
+}
+
+// Deserialize just the subset of fields we need to avoid breaking changes
+type sbomMetadata struct {
+	BOM *platform.LayerMetadata `json:"sbom" toml:"sbom"`
+}
+
+const (
+	Local = iota
+	Remote
+)
+
+// DownloadSBOM pulls SBOM layer from an image.
+// It reads the SBOM metadata of an image then
+// pulls the corresponding diffId, if it exists
+func (c *Client) DownloadSBOM(name string, options DownloadSBOMOptions) error {
+	img, err := c.imageFetcher.Fetch(context.Background(), name, image.FetchOptions{Daemon: options.Daemon, PullPolicy: image.PullNever})
```

Comment:
Wouldn't this mean that it never actually pulls the image?

Reply:
It seems apt to me. `--local` is supposed to denote what's already on the daemon. If it did pull, then it risks conflating `--remote` and `--local`.

This pattern was borrowed from the following `inspect_image.go`, but happy to switch it if you have better ideas.

https://github.com/buildpacks/pack/blob/9efe05dbab113b8e35baf69377cc80a6eca92bfd/pkg/client/inspect_image.go#L88

Reply:
No, this is actually great. I had assumed that `--remote` would pull the image locally, which is why I was confused, but just getting it from the registry without pulling it is obviously much better 😄